### DISCUSSION TITLE
How do we update on a Single Image Deployment ?

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#6781] [Help]
I am running bb on a Single Image deployment using docker on a Mac (M1).
What is the correct way to update to the latest release ?
Hitting "Check For Updates" doesn't update my installation from V.1.1.5 to V.1.113

**Comment by melohagan** [comment]
Hi @poirazis 

If you used docker compose, then you can set the image to `latest` as per the example:
https://docs.budibase.com/docs/docker

The command would just be `docker-compose up -d`

**Reply by poirazis** [reply]
Hey @melohagan 
I used the Method 1 in that link to install, not the docker compose one.

**Reply by melohagan** [reply]
I found a guide here: https://phoenixnap.com/kb/update-docker-image-container

I think you just need to pull the latest image, remove the old container and then spin up a new one. 

I think from my own experience your data should be saved in a volume so there shouldn't be any loss of data **although I'm not absolutely certain about that.**

**Reply by mike12345567** [reply]
Hi @poirazis - what @melohagan has said is correct, you'll need to pull the latest version of the image, then create a new container with this image (the docker compose method simplifies this a bit).

Your data will be preserved assuming you mount the same directory to the same place on the new container. It is important to make sure that you have shut down the old container before you start the new one as running two Budibase systems on the same volume could cause problems. It may be worth taking a copy of the local directory containing the volume data after shutting down the old container just to make sure you have a backup if needed.

The docker compose method simply takes a lot of this work out as compose itself will handle shutting down containers, pulling new images and remounting everything to a new container.

EDIT: you also raise a good point about the updates screen, we need to hide this as it can only work for our docker compose method, we should hide the update option in the UI when it cannot be performed.

**Reply by poirazis** [reply]
Hey @mike12345567 
I understand the process now. Maybe at some point in the future this would be taken care of by the CLI. (stopping the container / removing the container / spinning up a new container with the latest image keeping the same name and variables / path ) for Single Image installations.

I will revert to the docker compose installation for ease of use.

Thank you both, and have a blessed day!

**Reply by mike12345567** [reply]
Hi @poirazis - yes thats a good idea, hopefully will get a chance at some point to add some single image controls/backup functionality to the CLI!
